site/_guides/gatekeeper.md

```diff
@@ -0,0 +1,406 @@
+```
+httpproxy.projectcontour.io/demo created
+Error from server ([denied by httpproxy-unique-fqdn] HTTPProxy must have a unique FQDN): error when creating "httpproxies.yml": admission webhook "validation.gatekeeper.sh" denied the request: [denied by httpproxy-unique-fqdn] HTTPProxy must have a unique FQDN
+```
+
+The first `HTTPProxy` was created successfully, because there was not already an existing proxy with the `demo.projectcontour.io` FQDN.
+However, when the second `HTTPProxy` was submitted, Gatekeeper rejected its creation because it used the same FQDN as the first one.
+
+### Configure HTTPProxy policies
+
+The next constraint template and constraint that we'll create are what we refer to as a **policy**.
+These are rules for `HTTPProxy` specs that an individual Contour administrator may want to enforce for their cluster, but that are not explicitly required by Contour itself.
+In this example, we'll define a constraint template and constraint to enforce that all `HTTPProxies` can be configured with at most five retries for any route.
+
+Create a file called `retry-count-range-template.yml` containing the following YAML:
+
+```yaml
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: httpproxyretrycountrange
+spec:
+  crd:
+    spec:
+      names:
+        kind: HTTPProxyRetryCountRange
+        listKind: HTTPProxyRetryCountRangeList
+        plural: HTTPProxyRetryCountRanges
+        singular: HTTPProxyRetryCountRange
+      scope: Namespaced
+      validation:
+        openAPIV3Schema:
+          properties:
+            min:
+              type: integer
+            max: 
+              type: integer
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package httpproxy.retrycountrange
+
+        violation[{"msg": msg}] {
+          got := input.review.object.spec.routes[_].retryPolicy.count
+
+          got > input.parameters.max
+          msg := sprintf("retry policy count must be less than/equal to %v", [input.parameters.max])
+        }
+
+        violation[{"msg": msg}] {
+          got := input.review.object.spec.routes[_].retryPolicy.count
+
+          got < input.parameters.min
+          msg := sprintf("retry policy count must be greater than/equal to %v", [input.parameters.min])
+        }
+```
+
+Apply it to the cluster:
+
+```bash
+$ kubectl apply -f retry-count-range-template.yml
+```
+
+Again, within a few seconds, you'll see that a corresponding CRD has been created in the cluster:
+
+```bash
+$ kubectl get crd httpproxyretrycountrange.constraints.gatekeeper.sh
+NAME                                                 CREATED AT
+httpproxyretrycountrange.constraints.gatekeeper.sh   2020-08-13T16:12:10Z
+```
+
+Now, create a file called `retry-count-range-constraint.yml` containing the following YAML:
+
+```yaml
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: HTTPProxyRetryCountRange
+metadata:
+  name: httpproxy-retry-count-range
```

Comment:
I think the example should specify `metadata.namespace` field explicitly and a comment below should make the scoping of the constraint extra clear. In other words, if I create this `HTTPProxyRetryCountRange` in namespace `foo`, does it only apply to `HTTPProxy` objects in that namespace? Does it depend on the content of the policy?